src/riak_kv_qry_compiler.erl

```diff
@@ -147,19 +147,18 @@ compile_where_clause(?DDL{} = DDL,
 %% now break out the query on quantum boundaries
 -spec expand_query(?DDL{}, ?SQL_SELECT{}, proplists:proplist()) ->
                           {ok, [?SQL_SELECT{}]} | {error, term()}.
-expand_query(?DDL{table = Table, local_key = LK, partition_key = PK},
-             ?SQL_SELECT{} = Q1, Where1) ->
+expand_query(?DDL{local_key = LK, partition_key = PK},
+             ?SQL_SELECT{helper_mod = Mod} = Q1, Where1) ->
     case expand_where(Where1, PK) of
         {error, E} ->
             {error, E};
         {ok, Where2} ->
-            Mod = riak_ql_ddl:make_module_name(Table),
-            IsDescending = lists:member(descending, Mod:field_orders()),
+            IsDescending = is_partition_key_cols_descending(Mod:field_orders(), PK),
             SubQueries1 =
                 [Q1?SQL_SELECT{
                        is_executable = true,
                        type          = timeseries,
-                       'WHERE'       = maybe_fix_start_order(IsDescending, X, Mod, LK),
+                       'WHERE'       = maybe_fix_start_order(IsDescending, X),
```

Comment:
It seems you are applying `maybe_fix_start_order` for columns are not declared DESC in the DDL. Is it correct?